xarray/tests/test_backends.py

```diff
@@ -1264,6 +1268,24 @@ def test_compressor_encoding(self):
         with self.roundtrip(original, save_kwargs=save_kwargs) as actual:
             assert actual.var1.encoding['compressor'] == blosc_comp
 
+    def test_group(self):
+        original = create_test_data()
+        group = 'some/random/path'
+        with self.roundtrip(original, save_kwargs={'group': group},
+                            open_kwargs={'group': group}) as actual:
+            self.assertDatasetIdentical(original, actual)
+        with pytest.raises(KeyError):
+            with self.roundtrip(original,
+                                save_kwargs={'group': group}) as actual:
+                self.assertDatasetIdentical(original, actual)
+        # if we open the dataset without specifying group but with mode='r+',
+        # no error is raised, but the variables are not there. This is because
+        # xarray creates the attribute it needs. Is this the right behavior?
+        # Should we even allow 'r+' mode?
+        with self.roundtrip(original, save_kwargs={'group': group},
+                            open_kwargs={'mode': 'r+'}) as actual:
+            assert len(actual.variables) == 0
```

Comment:
Does anyone have an opinion about allowing `open_zarr(mode='r+')`. According to zarr docs:

>  ‘r+’ means read/write (must exist)

I'm not sure if read/write access actually makes sense for xarray. Is there any way for xarray to actually update the values of a datastore that has already been opened? If so, I'm not aware of it. It seems like we either `open_zarr` or call `to_zarr`. In the former case, only `r` access makes sense, and in the latter, only `w` / `w-` access makes sense.

Thoughts on this?

Reply:
Xarray never opens datastores for both reads and writes. So there's no point in supporting `r+` for now.